cose/key_eddsa.go

```diff
@@ -0,0 +1,81 @@
+package cose
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/ed25519"
+	"fmt"
+
+	"github.com/fxamacker/cbor/v2"
+)
+
+// An EdDSAPublicKey is a public key using EdDSA.
+type EdDSAPublicKey struct {
+	key ed25519.PublicKey
+}
+
+// UnmarshalEdDSAPublicKey unmarshals an EdDSA key using the COSE_Key format.
+func UnmarshalEdDSAPublicKey(raw []byte) (key *EdDSAPublicKey, remaining []byte, err error) {
```

Comment:
this is probably a me thing, but I find the named result params here slightly harder for me to reason about (explicit nil on final return, `remaining` declared relatively far away from use). 

FWIW, I didn't find any explicit guidance on this so take it or leave it. 

- https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_avoid_named_return_values
- https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

Reply:
the purpose for the named result parameters is documentation. Otherwise the meaning of `[]byte` would be unclear.

Reply:
Thanks for clarification. LGTM then.